tests/unit/test_charm.py

```diff
@@ -82,761 +137,432 @@ def _read_file(path: str) -> str:
-        self, _, patch_nrf_url, patch_check_output, patch_get_assigned_certificates
+        self, add_storage, fiveg_nrf_relation_id, certificates_relation_id
     ):
-        self.harness.add_storage(storage_name="certs", attach=True)
-        self.harness.add_storage(storage_name="config", attach=True)
-
-        root = self.harness.get_filesystem_root(self.container_name)
-        self._create_nrf_relation()
-        patch_check_output.return_value = POD_IP
-        patch_nrf_url.return_value = VALID_NRF_URL
-        cert_rel_id = self._create_certificates_relation()
+        root = self.harness.get_filesystem_root(CONTAINER_NAME)
 
         provider_certificate = Mock(ProviderCertificate)
         provider_certificate.certificate = STORED_CERTIFICATE
         provider_certificate.csr = STORED_CSR
-        patch_get_assigned_certificates.return_value = [provider_certificate]
+        self.mock_get_assigned_certificates.return_value = [provider_certificate]
 
         (root / CSR_PATH).write_text(STORED_CSR.decode())
         (root / CERT_PATH).write_text(STORED_CERTIFICATE)
         (root / CONFIG_PATH).write_text("super different config file content")
 
-        self.harness.remove_relation(cert_rel_id)
-        self.harness.set_can_connect(container=self.container_name, val=False)
+        self.harness.remove_relation(certificates_relation_id)
+        self.harness.set_can_connect(container=CONTAINER_NAME, val=False)
         self.harness.evaluate_status()
-        self.assertEqual(
-            self.harness.charm.unit.status, WaitingStatus("Waiting for container to start")
-        )
+        assert self.harness.charm.unit.status == WaitingStatus("Waiting for container to start")
 
-    def test_given_nrf_data_not_available_when_pebble_ready_then_status_is_waiting(self):
-        self.harness.add_relation(relation_name=NRF_RELATION_NAME, remote_app="some_nrf_app")
-        self._create_certificates_relation()
-        self.harness.container_pebble_ready(self.container_name)
+    def test_given_nrf_data_not_available_when_pebble_ready_then_status_is_waiting(
+        self, fiveg_nrf_relation_id, certificates_relation_id
+    ):
+        self.mock_nrf_url.return_value = ""
+        self.harness.container_pebble_ready(CONTAINER_NAME)
         self.harness.evaluate_status()
-        self.assertEqual(
-            self.harness.model.unit.status, WaitingStatus("Waiting for NRF data to be available")
-        )
+        assert self.harness.model.unit.status == WaitingStatus("Waiting for NRF data to be available")  # noqa: E501
 
-    @patch(f"{CERTIFICATES_LIB}.get_assigned_certificates")
-    @patch("charm.check_output")
-    @patch("charms.sdcore_nrf_k8s.v0.fiveg_nrf.NRFRequires.nrf_url", new_callable=PropertyMock)
-    @patch("ops.model.Container.restart")
     def test_given_relation_created_and_nrf_data_available_and_config_storage_not_attached_when_pebble_ready_then_status_is_waiting(  # noqa: E501
```

Comment:
this and `test_given_relations_created_and_nrf_data_available_and_certs_storage_not_attached_when_pebble_ready_then_status_is_waiting` could be a parametrized test based on the storage name `certs`, `config`